map-low.go

```diff
@@ -279,9 +279,17 @@ func (m *BPFMapLow) GetValueFlags(key unsafe.Pointer, flags MapFlag) ([]byte, er
 	return value, nil
 }
 
-// TODO: implement `bpf_map__lookup_and_delete_elem`
-// func (m *BPFMapLow) GetValueAndDeleteKey(key unsafe.Pointer) ([]byte, error) {
-// }
+// GetValueAndDeleteKey gets the value with the given key and delete the key from the map.
+// It returns the value and delete the given key.
+func (m *BPFMapLow) GetValueAndDeleteKey(key unsafe.Pointer) ([]byte, error) {
+	value := make([]byte, m.ValueSize())
```

Comment:
In `GetValueFlags` above, we took into account calculating the value size via `calcMapValueSize`, even this being only done by the `bpf_map__lookup_elem` version (which in turn calls `bpf_map_lookup_elem_flags`.

I think it's good to have this pre-calculation as a previous step in the upcoming map API as well.

Reply:
> In `GetValueFlags` above, we took into account calculating the value size via `calcMapValueSize`, even this being only done by the `bpf_map__lookup_elem` version (which in turn calls `bpf_map_lookup_elem_flags`.
> 
> I think it's good to have this pre-calculation as a previous step in the upcoming map API as well.

hi, deyslan, add these apis for look_and_delete, is it ok?
`func (m *BPFMapLow) GetValueAndDeleteKeyFlags(key unsafe.Pointer, flags MapFlag)`
`bpf_map_lookup_and_delete_elem` 

`func (m *BPFMapLow) GetValueAndDeleteKey(key unsafe.Pointer)`
`bpf_map_lookup_and_delete_elem_flags` 

`func (m *BPFMap) GetValueAndDeleteKey(key unsafe.Pointer, flags MapFlag)`
`bpf_map__lookup_and_delete_elem`

Reply:
You can create both, as we already do in GetValue{Flags}, setting the default flag for the shorter version and explaining well in the comments the shorter signature behaviour.

![image](https://github.com/aquasecurity/libbpfgo/assets/3372117/c32d5dda-224c-40eb-97b0-fcd5fd90c827)

Just make sure of wrapping the correct version (for BPFMap and BPFMapLow) as I mentioned [above](https://github.com/aquasecurity/libbpfgo/pull/411/files#r1546773498).

Reply:
done